src/types/widget.ts

```diff
@@ -394,6 +398,12 @@ export type Widget<
     | 'ais.stats'
     | 'ais.toggleRefinement'
     | 'ais.voiceSearch';
+
+  /**
+   * private marker on widgets to differentiate a widget from a connector
+   */
+  $$official?: boolean;
```

Comment:
Why can't we base this on the `ais` namespace? (`$$type.startsWith('ais.')`).

It'd be nice not having to extend our APIs for this.

Reply:
the goal is to differentiate between a widget and a connector. Do you see another way?

Reply:
going with $widgetType in a separate PR

Reply:
https://github.com/algolia/instantsearch.js/pull/4624